### PR TITLE
modified merging scheme and tighter outlier suppression  in primary vertex reconstruction

### PR DESCRIPTION
**Pull request by werdmann**
#### PR description:
The 3d clustering is modified slightly in order to reduce splitting and fake rates for Run 3. This addresses fake-rate and splitting related observations made in Run 2 data. Details and results can be found in 
https://indico.cern.ch/event/1019373/contributions/4293207/attachments/2217496/3754437/2021-03-29-tuning.pdf
and 
https://indico.cern.ch/event/1023338/contributions/4298206/attachments/2219781/3758715/2021-04-01-pvtuning.pdf
The code changes are
1) remove the Tc control for merging. The net effect of this feature is an increased fake rate that does not justify the small gain in efficiency. Updates of Tc are now initiated by the split() routine.
2) the proporionality constant of the outlier suppression term is increased from 1/#tracks to 1/#clusters. The value of dzCutOff can now be seen as the soft z-cut-off for an average cluster.

Small changes in the number and composition of reconstructed vertices are expected.

#### PR validation:
The code was tested with several 11_3_0_pre4 relval samples with the results reported in the presentations mentioned above.
runTheMatrix with this PR code on top of CMSSW_11_3_X_2021-04-05-2300 produces some errors, however, I can not see any connection to the vertex reconstruction  ( 37 34 26 18 14 4 1 1 1 tests passed, 0 2 7 0 0 0 0 0 0 failed )

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by mtosi** [comment]
@werdmann thanks for this PR, may I suggest updating the title and adding `[PV reco]` or something like that ?

thanks a lot !

**Comment by perrotta** [comment]
> @werdmann thanks for this PR, may I suggest updating the title and adding `[PV reco]` or something like that ?
> 
> thanks a lot !

@mtosi can I consider this as an approval of the update of the algorithm from the Tracking POG?
As far as I can see, the outputs of the (low stat) jenkins tests do not show issues that can contradict what was presented and discussed in the POG

File: RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc [hunk 1]
```diff
@@ -483,31 +483,26 @@ bool DAClusterizerInZ_vect::merge(vertex_t& y, track_t& tks, double& beta) const
   for (unsigned int ik = 0; ik < critical.size(); ik++) {
     unsigned int k = critical[ik].second;
     double rho = y.rho[k] + y.rho[k + 1];
-    double swE = y.swE[k] + y.swE[k + 1] - y.rho[k] * y.rho[k + 1] / rho * std::pow(y.zvtx[k + 1] - y.zvtx[k], 2);
-    double Tc = 2 * swE / (y.sw[k] + y.sw[k + 1]);
```
[the remark this answers is not in the document]
Comment (werdmann) [reply RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc:487]: thanks for catching. I'll also modify the "dump" routine, which is only used when DEBUG is enabled, to print meaningful Tc.

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Small modifications to the algorithm, implemented as stated in the PR description.
- Jenkins tests show overall a slight reduction in the number of reconstructed vertices (expected, with the reduction of fakes and splitting) and a few minor changes in all related quantities. Larger stat validation is available in the plots attached to the PR description.
- Update discussed and approved in the tracking POG.

**Comment by silviodonato** [comment]
+1

**Pull request merged**